nautobot/extras/tests/test_customfields.py

```diff
@@ -385,15 +385,51 @@ def test_to_filter_field(self):
 
 class CustomFieldManagerTest(TestCase):
     def setUp(self):
-        content_type = ContentType.objects.get_for_model(Location)
+        self.content_type = ContentType.objects.get_for_model(Location)
         custom_field = CustomField(type=CustomFieldTypeChoices.TYPE_TEXT, label="Text Field", default="foo")
         custom_field.save()
-        custom_field.content_types.set([content_type])
+        custom_field.content_types.set([self.content_type])
 
     def test_get_for_model(self):
         self.assertEqual(CustomField.objects.get_for_model(Location).count(), 2)
         self.assertEqual(CustomField.objects.get_for_model(VirtualMachine).count(), 0)
 
+    def test_get_for_model_lru_cache(self):
+        """Test that the lru cache is properly invalidated when CustomFields are created or deleted."""
+
+        qs1 = CustomField.objects.get_for_model(Location)
+        self.assertIsNone(qs1._result_cache)
+        list(qs1)  # force queryset evaluation
+        self.assertIsNotNone(qs1._result_cache)
```

Comment:
This is surprising to me that calling get_for_model doesn't populate the cache, only evaluating the queryset does. I guess this is an interaction between lru_cache and Django's lazy queryset evaluation? It feels a bit weird to do the initial assertIsNone as that feels like an implementation detail. Perhaps?

```suggestion
        qs1 = CustomField.objects.get_for_model(Location)
        self.assertEqual(qs1.count(), 2)
        self.assertIsNotNone(qs1._result_cache)
```

Actually I'm not a big fan of checking `_result_cache` either, would it be possible/better to check the return from `qs1.cache_info()` instead?

Similar comments on the other checks below as well.

Reply:
> This is surprising to me that calling get_for_model doesn't populate the cache, only evaluating the queryset does

I'm just evaluating the queryset so that I can see if the cache was used but I *might* be able to just say `qs1 is qs2` to determine if the cache is being used?

Reply:
> Actually I'm not a big fan of checking _result_cache either, would it be possible/better to check the return from qs1.cache_info() instead?

I'm not seeing a `cache_info` anywhere in Django